src/app/shared/services/token.service.ts

```diff
@@ -30,13 +30,21 @@ export class TokenService {
 
   checkTokenExpiration() {
     const currentTime = new Date().toISOString();
-    const tokenExpirationTime = localStorage.getItem("civis-token_expires") ?? new Date(0).toISOString();
-   
-    // logout user if token expired
-    if (currentTime > tokenExpirationTime) {
-      localStorage.removeItem("civis-token");
-      localStorage.removeItem("civis-token_expires");
-    } 
+    const tokenExpirationString = localStorage.getItem("civis-token_expires");
+
+    const tokenExpirationTime = new Date(tokenExpirationString);
```

Comment:
Before this add these 2 validations:
1. `tokenExpirationString` is a `string`
2. `tokenExpirationString` is a valid date string. Use regex for this